sql-data-pipeline/src/main/java/io/cdap/pipeline/sql/app/bigquery/BigQueryExecutor.java

```diff
@@ -0,0 +1,332 @@
+    }
+    // Schema and framework config creation
+    SchemaPlus rootSchema = CalciteSchema.createRootSchema(true).plus();
+    final FrameworkConfig builderConfig = Frameworks.newConfigBuilder().defaultSchema(rootSchema).build();
+    // Identifier for temporary table
+    int temporaryTableCounter = 0;
+    // Traverse the topological ordering
+    for (String node: topologicalOrder) {
+      LOG.info("Traversing Node: " + node);
+      // Get the plugin
+      SQLTransform plugin = pluginMap.get(node);
+
+      // Create the RelBuilder for generating the query
+      final RelBuilder builder = RelBuilder.create(builderConfig);
+
+      // Generate schema from source nodes
+      // Adding the tables directly to the SchemaPlus seems kind of hacky
+      // TODO: Revisit schema generation
+      if (sourceStages.contains(node)) {
+        // The stage is a source node
+        SQLSource sourcePlugin = (SQLSource) plugin;
+        AbstractTableInfo table = sourcePlugin.getSourceTable();
+        // Add the table to the root schema
+        rootSchema.add(table.getTableName(), table);
+        // Add a scan for the source table
+        builder.scan(table.getTableName());
+      } else {
+        // Node is not a source node
+        // Add all dependencies of the current node to the builder parameter stack
+        Set<String> fromNodes = fromNodeMap.get(node);
+        for (String fromNode : fromNodes) {
+          // Check if there exists a temporary table
+          if (temporaryTableMap.containsKey(fromNode)) {
+            // Scan the temporary table
+            builder.scan(temporaryTableMap.get(fromNode));
+          } else {
+            // Otherwise push the inlined query
+            builder.push(relationalNodeMap.get(fromNode));
+          }
+        }
+
+        // Pass the builder to the node to allow it to combine the inputs
+        int dependencies = fromNodes.size();
+        plugin.combineInputs(builder, dependencies);
+      }
+
+      // Get the relational query node and set the node in the map
+      relationalNodeMap.put(node, plugin.getQuery(builder));
+
+      // Check if the node is a splitter node
+      if (toNodeMap.containsKey(node) && toNodeMap.get(node).size() > 1) {
+        // Add a temporary table if splitter
+        String temporaryTableName = TEMPORARY_TABLE_PREFIX + temporaryTableCounter++;
+        temporaryTableMap.put(node, temporaryTableName);
+        // This adds the potential temporary table schema to the root schema
+        TemporaryTable temporaryTable = new TemporaryTable(temporaryTableName,
+                                                           relationalNodeMap.get(node).getRowType());
+        rootSchema.add(temporaryTableName, temporaryTable);
+      }
+    }
+
+    // Serialize all sink and splitter RelNodes
+    // Converter to convert RelNodes to SqlNodes
+    RelToSqlConverter converter = new RelToSqlConverter(BigQuerySqlDialect.DEFAULT);
```

Comment:
Seems only this part is about big query, it is better we can have an `AbstractExecutor` which handles the above logic, so that if we want to support other sql types, we can just extend from that class, and we just need to implement the sql specific logic.

Reply:
Fixed, created a separate AbstractExecutor class for other SQL types to extend.